### DISCUSSION TITLE
Renovate ignores new versions of docker image

### DISCUSSION THREAD

**Discussion by wrbbz** [renovatebot/renovate#20463] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

32.31.1

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### If you're self-hosting Renovate, tell us what version of the platform you run.

GitLab EE 15.7

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Hello!
I'm trying to make renovate to update an image in .gitlab-ci.yml file and have no luck in it.
I would like renovate to make a PR with new version of the image, but it ignores new versions

### What you tried so far.

I've tried different configurations and decided to use almost the default one:
```json
{
  "extends": [
    "config:base"
  ],
  "gitlabci": {
    "fileMatch": [
      "\\.gitlab-ci\\.yml$"
    ],
    "pinDigests": true,
    "automerge": true
  }
}
```

The image in .gitlab-ci.yml is stated as a global value:

```yaml
image: ghcr.io/gitlabform/gitlabform:2.12.0-alpine3.14
```

With this config renovate successfully pined the dependency and made a PR with that change.
Nevertheless, it does not updates the dependency to a new version.

### Relevant debug logs

Renovate successfully finds the dependency and its repo, but does not use the latest version

<details><summary>Logs</summary>

```json
       "config": {
         "gitlabci": [
           {
             "packageFile": ".gitlab-ci.yml",
             "deps": [
               {
                 "depName": "ghcr.io/gitlabform/gitlabform",
                 "currentValue": "2.12.0-alpine3.14",
                 "currentDigest": "sha256:d529db367459aceb44d9d1ef8c569b69921a52ce930457010a550f2b60cc5ec5",
                 "replaceString": "ghcr.io/gitlabform/gitlabform:2.12.0-alpine3.14@sha256:d529db367459aceb44d9d1ef8c569b69921a52ce930457010a550f2b60cc5ec5",
                 "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
                 "datasource": "docker",
                 "depType": "image",
                 "depIndex": 0,
                 "updates": [],
                 "warnings": [],
                 "versioning": "docker",
                 "sourceUrl": "https://github.com/gitlabform/gitlabform",
                 "currentVersion": "2.12.0",
                 "fixedVersion": "2.12.0-alpine3.14"
               }
             ]
           }
         ]
       }
```

</details>

**Comment by wrbbz** [comment, marked as answer]
Here are some requests to ghcr.io (from debug log)
```
         "https://ghcr.io/token (GET,200)": 1,
         "https://ghcr.io/v2/ (GET,401)": 1,
         "https://ghcr.io/v2/gitlabform/gitlabform/manifests/2.12.0-alpine3.14 (HEAD,200)": 1,
         "https://ghcr.io/v2/gitlabform/gitlabform/tags/list (GET,200)": 1
```

And here is the [link](https://github.com/gitlabform/gitlabform/pkgs/container/gitlabform) to the gitlabform tags


### UPDATE

I'll post the solution here. For future (and mark it as answered)

There was a problem not with Renovate configuration. Container registry does not have the newest version with `alpine` tag. Hence, a PR was not created. After I changed the tag to `2.12.0` instead of `2.12.0-alpine3.14` a PR was created successfully

**Reply by rarkins** [reply]
What is the newer version you expect it to update to?

**Reply by wrbbz** [reply]
At least - 3.0.0, but the latest is 3.4.1

**Reply by rarkins** [reply]
What is the exact tag

**Reply by wrbbz** [reply]
I've added `matchUpdateTypes` to config:
```json
 {
   "extends": [
     "config:base"
   ],
   "gitlabci": {
     "fileMatch": [
       "\\.gitlab-ci\\.yml$"
     ],
    **"matchUpdateTypes": ["major", "minor", "patch"]**
    "automerge": true
   }
 }
 ```
 And that resulted to an additional requests to ghcr.io and the following output:
 ```
 DEBUG: found labels in manifest (repository=${analysedRepo})
       "labels": {
         "org.opencontainers.image.created": "2023-01-17T12:50:51.875Z",
         "org.opencontainers.image.description": "🏗 Specialized configuration as a code tool for GitLab",
         "org.opencontainers.image.licenses": "MIT",
         "org.opencontainers.image.revision": "9563fd6ed5d3023de9c6c14b8fa2a57c1cef44a2",
         "org.opencontainers.image.source": "https://github.com/gitlabform/gitlabform",
         "org.opencontainers.image.title": "gitlabform",
         "org.opencontainers.image.url": "https://github.com/gitlabform/gitlabform",
         "org.opencontainers.image.version": "3.4.1"
       }
 ```
 And the requests are:
 ```
"https://ghcr.io/token (GET,200)": 1,
"https://ghcr.io/v2/ (GET,401)": 1,
"https://ghcr.io/v2/gitlabform/gitlabform/blobs/sha256:d9c31701f44a6a2a5d673bd7ea5eb0721d5624748805416d6fb76e8c7dcc26f1 (GET,200)": 1,
"https://ghcr.io/v2/gitlabform/gitlabform/manifests/2.12.0-alpine3.14 (HEAD,200)": 1,
"https://ghcr.io/v2/gitlabform/gitlabform/manifests/latest (GET,200)": 1,
"https://ghcr.io/v2/gitlabform/gitlabform/manifests/sha256:f0573d3d3ee410806c5c2661b5ab28b1252ebeb0d3a957111ce6b26f172c21a9 (GET,200)": 1,
"https://ghcr.io/v2/gitlabform/gitlabform/tags/list (GET,200)": 1,
```

But PR was never created

**Reply by wrbbz** [reply]
> What is the exact tag

`ghcr.io/gitlabform/gitlabform:3.4.1@sha256:9be49923a9d9fd48846ef99d42f390e7f6304c7f94b8ff4b9d0dfbf92039d8ea`

**Reply by rarkins** [reply]
So you want to update from `2.12.0-alpine3.14` to `3.4.1`?

Renovate won't do that by default because for most images if there's a `-alpine3.14` tag it won't be compatible with a tag with no alpine suffix.

For this you need to configure Renovate to use a different versioning. You can read about default Docker versioning at https://docs.renovatebot.com/modules/versioning/#docker-versioning

**Reply by wrbbz** [reply]
Thats true. My bad. I absolutely missed that there is no `alpine` tag in `3.4.1`
Changed image from `2.12.0-alpine3.14` to `2.12.0` and PR was created. Thank you!

**Comment by HonkingGoose** [comment]
> ### If you're self-hosting Renovate, tell us what version of Renovate you run.
> 
> 32.31.1

By the way, you're two major versions behind. `34.x.x` is the latest major release. I don't know if updating Renovate fixes your problem.

I'll let the maintainers respond to your questions. 😉

**Reply by wrbbz** [reply]
The same with version 34.143.0

**Reply by HonkingGoose** [reply]
Thank you for trying the newest version. ❤️
